python/cudf/cudf/_lib/pylibcudf/strings/regex_program.pyx

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+
+from libcpp.memory cimport unique_ptr
+from libcpp.string cimport string
+from libcpp.utility cimport move
+
+from cudf._lib.pylibcudf.libcudf.strings.regex_flags cimport regex_flags
+from cudf._lib.pylibcudf.libcudf.strings.regex_program cimport regex_program
+
+from cudf._lib.pylibcudf.strings.regex_flags import RegexFlags
+
+
+cdef class RegexProgram:
+
+    def __init__(self, *args, **kwargs):
+        raise ValueError("Do not instantiate RegexProgram directly, use create")
+
+    @staticmethod
+    def create(object pattern, object flags):
```

Comment:
```suggestion
    def create(str pattern, RegexFlags flags):
```
?

Reply:
My experimentation leads me to believe it must be done this way: https://github.com/rapidsai/cudf/pull/15880/commits/758755c33c78e50ad5b7c48e7e54eac682c47e1f

Reply:
Yes, you always have to use the `cimport`ed name when typing Cython functions, not the imported name. `regex_flags` != `RegexFlags` because the latter is imported rather than cimported. This is deliberate! We import the aliased name because that is the name we want to expose to users of pylibcudf in pure Python, using Python naming conventions for classes (CapsCase) since enums in Python are classes, while `regex_flags` is the snake_case representation of the Cython/C enum.